troposphere/static/js/components/modals/volume/VolumeAttachModal.react.js

```diff
@@ -56,28 +56,31 @@ export default React.createClass({
 
         var project = this.props.project;
 
-        // TODO: remove ambiguity between state/this.state
-        var state = {
+        // prepare to build out the `next` state for the component
+        var next = {
             instances: stores.ProjectInstanceStore.getInstancesFor(project),
             instanceId: null
         };
 
         this.state = this.state || {};
 
-        // Use selected instance or default to the first one
-        if (state.instances) {
+        // Use selected instance or default to the first one (when available)
+        if (next.instances) {
           var volume = this.props.volume,
-            InstanceCollection = state.instances.constructor;
+              firstInstanceId = null,
+              InstanceCollection = next.instances.constructor;
 
           // Filter out instances not in the same provider as the volume
-          state.instances = state.instances.filter(function (i) {
+          next.instances = next.instances.filter(function (i) {
             return i.get('identity').provider === volume.get('identity').provider;
           });
-          state.instances = new InstanceCollection(state.instances);
-          state.instanceId = this.state.instanceId || state.instances.first().id;
+          next.instances = new InstanceCollection(next.instances);
```

Comment:
This is just a suggestion. This component is due for a make over, i'm okay with the set of changes introduced here, and a refactor can address it later.